internal/compiler/engine.go

```diff
@@ -66,3 +74,60 @@ func (e *Engine) CompileString(input string, context map[string]any) (string, er
 	}
 	return strings.TrimSpace(buf.String()), nil
 }
+
+func (e *Engine) CompileAssets(ctx context.Context, startTime, endTime time.Time, configs, assets, systemEnvVars map[string]string) (map[string]string, error) {
```

Comment:
Can we move this to a domain specific place, or near other compile assets.

Reply:
move the granular changes for compile assets here https://github.com/goto/optimus/pull/133